### DISCUSSION TITLE
in QField is there a variable to retrieve the altitude: (@position_quotageid) obtained from the vertical grid shit in use

### DISCUSSION THREAD

**Discussion by Desaxo** [opengisch/QField#3990] [Q&A]
Hello Qfield Teams, 

I can easily retrieve X and Y coordinates using the variables:
/*x($geometry)*/ and /*Y($geometry).*/

On the other hand, I cannot retrieve the corrected altitude which is
displayed on the Qfield coordinate tracking. If I use for the altitude Z
the variable/*z($geometry)*/, I have a "Null" value. If I use
/*z(position_coordinate)*/ I have the altitude relative to the Ellipsoid
in WGS 84.

My problem is to be able to retrieve the altitude corrected by Qfield
and which is displayed in the field on my image.

In which variable is this displayed value stored?

And how can I assign it to fields in my form prepared on Qgis?

Cordially JD

<img width="600" alt="Screenshot_2" src="https://user-images.githubusercontent.com/86613712/222220753-33cfe273-e6ad-4eaf-83e5-1acc234db4f2.png">
<img width="1243" alt="QField_attribut" src="https://user-images.githubusercontent.com/86613712/222220997-63ca0627-9c21-49ad-b1dd-5158a6560f3a.png">

**Comment by m-kuhn** [comment]
This is currently doable in the following way:

- make sure your layer geometry type is of type PointZ
- snap the cursor to your current position
- use `z($geometry)` as default value

There is room for improvement so the first condition on the layer geometry type is optional and to make it available via `@position_coordinate`.
